funcx_endpoint/funcx_endpoint/executors/high_throughput/executor.py

```diff
@@ -353,7 +355,18 @@ def __init__(
                 "--container_image={container_image} "
             )
 
-    def start(self, results_passthrough: multiprocessing.Queue = None):
+    def start(
+        self,
+        *args,
+        endpoint_id: t.Optional[uuid.UUID] = None,
+        run_dir: t.Optional[str] = None,
+        results_passthrough: t.Optional[multiprocessing.Queue] = None,
+        **kwargs,
+    ):
+        assert run_dir, "GCExecutor requires kwarg:run_dir at start"
+        assert endpoint_id, "GCExecutor requires kwarg:endpoint_id at start"
+        self.run_dir = run_dir
+        self.endpoint_id = endpoint_id
```

Comment:
If we're going to `assert` require them, why not tell typing that they're required?

Reply:
The base class defines start like this -> `GlobusComputeEngineBase.start(*args, **kwargs)`. These two options are not required for every executor, which is why it was left as optional in the base class, and so to have mypy not complain about liskov substitution, these args were left as optional.